go/libraries/doltcore/env/environment.go

```diff
@@ -432,37 +432,46 @@ func (dEnv *DoltEnv) configureRepo(doltDir string) error {
 }
 
 // Inits the dolt DB of this environment with an empty commit at the time given and writes default docs to disk.
-// Writes new repo state with a master branch and current root hash.
-func (dEnv *DoltEnv) InitDBAndRepoState(ctx context.Context, nbf *types.NomsBinFormat, name, email string, t time.Time) error {
-	err := dEnv.InitDBWithTime(ctx, nbf, name, email, t)
+// Writes new repo state with a main branch and current root hash.
+func (dEnv *DoltEnv) InitDBAndRepoState(ctx context.Context, nbf *types.NomsBinFormat, name, email, branchName string, t time.Time) error {
+	err := dEnv.InitDBWithTime(ctx, nbf, name, email, branchName, t)
 	if err != nil {
 		return err
 	}
 
-	return dEnv.InitializeRepoState(ctx)
+	return dEnv.InitializeRepoState(ctx, branchName)
 }
 
 // Inits the dolt DB of this environment with an empty commit at the time given and writes default docs to disk.
 // Does not update repo state.
-func (dEnv *DoltEnv) InitDBWithTime(ctx context.Context, nbf *types.NomsBinFormat, name, email string, t time.Time) error {
+func (dEnv *DoltEnv) InitDBWithTime(ctx context.Context, nbf *types.NomsBinFormat, name, email, branchName string, t time.Time) error {
 	var err error
 	dEnv.DoltDB, err = doltdb.LoadDoltDB(ctx, nbf, dEnv.urlStr, dEnv.FS)
 	if err != nil {
 		return err
 	}
 
-	err = dEnv.DoltDB.WriteEmptyRepoWithCommitTime(ctx, GetDefaultInitBranch(dEnv.Config), name, email, t)
+	initBranch := GetDefaultInitBranch(dEnv.Config)
```

Comment:
This is the wrong place to fill in a default. Make it the responsibility of clients to provide a branch name.